src/Microsoft.AspNetCore.Mvc.Core/ApplicationParts/IApplicationPartTypeProvider.cs

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace Microsoft.AspNetCore.Mvc.ApplicationParts
+{
+    /// <summary>
+    /// Gives access to a set of types from an <see cref="ApplicationPart"/>.
+    /// </summary>
+    public interface IApplicationPartTypeProvider
```

Comment:
Sorry, you are actually right, this will be needed as part of https://github.com/aspnet/Mvc/pull/4344. I'll move it to that PR.